### PR TITLE
Move run queue to SingleRunController

### PR DESCRIPTION
**Pull request by nsoblath**

### CODE REVIEW

**Comment by nsoblath** [comment]
@pslocum @psurukuc I shouldn't have done this, but I included two separate changes in one PR.  We can review them separately.  @pslocum could you please review the move of RunQueue into SingleRunController.  @psurukuc could you please review the move of ConfigException to its own file?

**Review by psurukuc — APPROVED** [review]

File: Cpp/Library/Control/SingleRunController.cc [hunk 1]
```diff
@@ -36,12 +38,125 @@ namespace Nymph
...
+        }
+        else
+        {
+            ConfigureRunQueue( node["run-queue"].as_array() );
+        }
+
+        return;
+    }
+
+    void SingleRunController::ConfigureRunQueue( const scarab::param_array& array )
```
Comment (pslocum) [inline Cpp/Library/Control/SingleRunController.cc:55]: Is this where the ordering of the processors gets arranged (e.g. series or parallel)?

Comment (nsoblath) [reply Cpp/Library/Control/SingleRunController.cc:55]: That's right.  The configuration could look something like this: `[proc1, proc2]` to run those two in series.  Or if it were configured with `[[proc1, proc2], proc3]`, then `proc1` and `proc2` would run in parallel, and then `proc3` would follow in series.

Comment (nsoblath) [reply Cpp/Library/Control/SingleRunController.cc:55]: This reminds me that I forgot to include the documentation.  That also needs to move over from ProcessorToolbox.

File: Cpp/Library/Control/SingleRunController.cc [hunk 2]
```diff
@@ -19,7 +18,10 @@ namespace Nymph
 {
     LOGGER( contlog, "SingleRunController");
 
-    SingleRunController::SingleRunController( const std::string& name ) :
+    SingleRunController::SingleRunController( const ProcessorToolbox& procTB, const std::string& name ) :
+            Controller(),
+            fProcTB( procTB ),
+            fRunQueue(),
```
[the remark this answers is not in the document]
Comment (nsoblath) [reply Cpp/Library/Control/SingleRunController.cc:24]: That's right

**Pull request merged**